src/mmds/src/lib.rs

```diff
@@ -254,6 +259,33 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_unsupported_http_method() {
+        // Test invalid HTTP POST method. This is not supported by `micro-http`.
+        let request = b"POST http://169.254.169.254/ HTTP/1.0\r\n\r\n";
+        let mut expected_response = Response::new(Version::Http10, StatusCode::NotImplemented);
+        expected_response.set_body(Body::new("Unsupported HTTP method.".to_string()));
+        let actual_response = parse_request(request);
+        assert!(expected_response.status() == actual_response.status());
+        assert!(expected_response.body().unwrap() == actual_response.body().unwrap());
```

Comment:
It looks like this code is duplicated several times where the only difference is the method. Can you make a vector of methods instead and iterate over them in a for?
```rust
let unsupported_methods = vec!["PATCH", "PUT", ....];
let mut expected_response = Response::new(Version::Http10, StatusCode::NotImplemented);
expected_response.set_body(Body::new("Unsupported HTTP method.".to_string()));
for method in unsupported_methods {
        let request = b"{} http://169.254.169.254/ HTTP/1.0\r\n\r\n".format(method);
        let actual_response = parse_request(request);
        assert!(expected_response.status() == actual_response.status());
        assert!(expected_response.body().unwrap() == actual_response.body().unwrap());
}```